### ISSUE TITLE
Clarify `--secs`, `--short` and `--equalize_test_with_non_secs`

### ISSUE THREAD

**Issue by L-M-Sherlock** [open-spaced-repetition/fsrs-benchmark#160]
The current behavior is:

- if `--short` is true, the same-day reviews are included in features but excluded in labels.

It's unintended if `--short` and `--secs` both are true. Because I also want to include fractional interval in labels when the `elapsed_days` is 0 but the `elapsed_seconds` is non-zero.

My intuition is:

- `--short` controls whether the same-day reviews are included in features and labels or not.
- `--secs` controls whether the interval of features and labels are fractional or not.

But it will induce new problem: if `--short` is true but `--secs` is false, the zero `delta_t` will be included in labels, which will induce errors in training.

@1DWalker do you have any idea to deal with it?

Btw, `--equalize_test_with_non_secs` enforces `delta_t` to non-fractional, which let the model use non-fractional interval to predict the retention. It's unintended.

**Comment by 1DWalker** [comment]
`--equalize_test_with_non_secs` will actually only set `delta_t` to be non-fractional so that RMSE (bins) functions properly in fsrs-optimizer. The model still uses fractional intervals via this variable that gets set
https://github.com/open-spaced-repetition/srs-benchmark/blob/0c3a2c818140fd52459792103d84a836530cb081/other.py#L2530

I think we should still give a model a way to only use same-day reviews in features and not use the labels. I think FSRS currently uses it this way(?)

We can make secs the default (no short). Add `--days` to convert fractional to integer for >= 1 day reviews. Then when `--short` is used it will naturally be with seconds since that was the normal default. We can also add `--short_features` if a model wants to only use features but not labels. I don't see why a model would want to use labels but not features but we can add `--short_labels` as well.

We would still need an equalization flag unless we want to split up the benchmark to non short and short with no possible comparison in between. `--short` with labels changes the size of the dataset. `--days` with the current outlier filter would also change it. The size of the dataset affects the TimeSplitSeries split that decides which reviews are used in the test set.

But, I'm against a large change like this. This seems like a lot of work for little reward. If someone has a specific set of behaviors that they want then they can write their own `create_features` function. The function already has a ton of casework for each individual model so it won't exactly be unprecedented.

**Comment by L-M-Sherlock** [comment]
> `--equalize_test_with_non_secs` will actually only set `delta_t` to be non-fractional so that RMSE (bins) functions properly in fsrs-optimizer. The model still uses fractional intervals via this variable that gets set

https://github.com/open-spaced-repetition/srs-benchmark/blob/0c3a2c818140fd52459792103d84a836530cb081/other.py#L2222-L2227

https://github.com/open-spaced-repetition/fsrs-optimizer/blob/3f9a8ee949acb2b375a79bbb60b5db416a81e260/src/fsrs_optimizer/fsrs_optimizer.py#L244-L265

The "delta_t" is used in here:

```python
        self.t_train = torch.tensor(dataframe["delta_t"].values, dtype=torch.float)
```

**Comment by 1DWalker** [comment]
Okay. Then to fix all these errors, we can introduce an "delta_t_days" that the RMSE (bins) function would use. Then "delta_t" would be free to be either fractional or integer.

**Comment by Expertium** [comment]
We really need to simplify this mess
How about just 3 commands?
`--delta_t_secs`. True = fractional intervals, False = integer intervals.
`--short_term_reviews`. True = use same-day reviews, False = don't.
`--short_term_loss`. True = calculate the loss on same-day reviews both in training **_and_** evaluation. False = never calculate the loss on same-day reviews, neither in training nor in evaluation.

**Comment by 1DWalker** [comment]
We can still have an equalization flag to able to compare `--short_term_loss` to current models, and introduce a new table for models that do predictions for all reviews.

**Comment by L-M-Sherlock** [comment]
> Okay. Then to fix all these errors, we can introduce an "delta_t_days" that the RMSE (bins) function would use. Then "delta_t" would be free to be either fractional or integer.

Here is the PR: https://github.com/open-spaced-repetition/srs-benchmark/pull/163

**Comment by L-M-Sherlock** [comment]
OK. I think we should enable `--equalize_test_with_non_secs` by default and remove it from the filename. Thoughts?

**Comment by 1DWalker** [comment]
Sure, but we should later on introduce a new flag that would use short-term labels in testing and add another table to the readme to benchmark short-term models.

**Comment by Expertium** [comment]
@L-M-Sherlock @1DWalker so what's the consensus on this:

> How about just 3 commands?
> --delta_t_secs. True = fractional intervals, False = integer intervals.
> --short_term_reviews. True = use same-day reviews, False = don't.
> --short_term_loss. True = calculate the loss on same-day reviews both in training and evaluation. False = never calculate the loss on same-day reviews, neither in training nor in evaluation.
